compiler/semantic/expr.go

```diff
@@ -93,14 +93,14 @@ func semExpr(scope *Scope, e ast.Expr) (dag.Expr, error) {
 		if err != nil {
 			return nil, err
 		}
-		typ, err := semType(scope, e.Type)
+		typ, err := semExpr(scope, e.Type)
 		if err != nil {
 			return nil, err
 		}
-		return &dag.Cast{
```

Comment:
Remove `dag.Cast` since this is the only place we create them.

Reply:
This was a little more involved than I thought!